astroquery/desi/core.py

```
@@ -0,0 +1,111 @@
+        """
+        Downloads the images for a certain region of interest.
+
+        Parameters
+        ----------
+        position: `~astropy.coordinates.SkyCoord`
+            coordinates around which we define our region of interest.
+        pixels: 'int', optional
+            pixel dimensions of the image to be produced.
+            Default to 60.
+        radius: `~astropy.units.Quantity`, optional
+            the radius of our region of interest.
+            Default to a value of 0.5 arcmin.
+        data_release: int, optional
+            the data release of the LegacySurvey to use.
+            Default to 9.
+        show_progress: bool, optional
+            whether to display a progress bar if the file is downloaded from a remote server.
+            Default to True.
+        image_band: str, optional
+            indicates the desired band.
+            Default to 'g'.
+
+        Returns
+        -------
+        list: A list of `~astropy.io.fits.HDUList` objects.
+        """
+
+        position_transformed = position.transform_to(coord.ICRS)
+
+        image_size_arcsec = radius.arcsec
+        pixsize = 2 * image_size_arcsec / pixels
+
+        image_url = (f"{conf.legacysurvey_service_url}?"
+                    f"ra={position_transformed.ra.deg}&"
+                    f"dec={position_transformed.dec.deg}&"
+                    f"size={pixels}&"
+                    f"layer=ls-dr{data_release}&"
+                    f"pixscale={pixsize}&"
+                    f"bands={image_band}")
+
+        file_container = commons.FileContainer(image_url, encoding='binary', show_progress=show_progress)
+
+        try:
+            fits_file = file_container.get_fits()
+        except (requests.exceptions.HTTPError, urllib.error.HTTPError) as exp:
+            fits_file = None
+            warnings.warn(f"{str(exp)} - Problem retrieving the file at the url: {image_url}", NoResultsWarning)
```

Following @bsipocz 's unresolved comment: if the download was unsuccessful, we should raise an Exception.
```suggestion
            raise urllib.error.HTTPError(f"{str(exp)} - Problem retrieving the file at the url: {image_url}")
```